src/Core/AdminConsole/OrganizationFeatures/Policies/Implementations/SavePolicyCommand.cs

```diff
@@ -0,0 +1,114 @@
+﻿#nullable enable
+
+using Bit.Core.AdminConsole.Entities;
+using Bit.Core.AdminConsole.Enums;
+using Bit.Core.AdminConsole.Repositories;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Services;
+
+namespace Bit.Core.AdminConsole.OrganizationFeatures.Policies.Implementations;
+
+public class SavePolicyCommand : ISavePolicyCommand
+{
+    private readonly IApplicationCacheService _applicationCacheService;
+    private readonly IEventService _eventService;
+    private readonly IPolicyRepository _policyRepository;
+    private readonly IReadOnlyDictionary<PolicyType, IPolicyDefinition> _policyDefinitions;
+
+    public SavePolicyCommand(
+        IApplicationCacheService applicationCacheService,
+        IEventService eventService,
+        IPolicyRepository policyRepository,
+        IEnumerable<IPolicyDefinition> policyDefinitions)
+    {
+        _applicationCacheService = applicationCacheService;
+        _eventService = eventService;
+        _policyRepository = policyRepository;
+
+        var policyDefinitionsDict = new Dictionary<PolicyType, IPolicyDefinition>();
+        foreach (var policyDefinition in policyDefinitions)
+        {
+            if (!policyDefinitionsDict.TryAdd(policyDefinition.Type, policyDefinition))
+            {
+                throw new Exception($"Duplicate PolicyDefinition for {policyDefinition.Type} policy.");
+            }
+        }
+
+        _policyDefinitions = policyDefinitionsDict;
+    }
+
+    public async Task SaveAsync(Policy policy, IOrganizationService organizationService, Guid? savingUserId)
+    {
+        var org = await _applicationCacheService.GetOrganizationAbilityAsync(policy.OrganizationId);
+        if (org == null)
+        {
+            throw new BadRequestException("Organization not found");
+        }
+
+        if (!org.UsePolicies)
+        {
+            throw new BadRequestException("This organization cannot use policies.");
+        }
+
+        var policyDefinition = GetDefinition(policy.Type);
+        var allSavedPolicies = await _policyRepository.GetManyByOrganizationIdAsync(org.Id);
+        var currentPolicy = allSavedPolicies.SingleOrDefault(p => p.Id == policy.Id);
```

Comment:
Correct. There's actually a uniqueness constraint over `Policy.OrganizationId` and `Policy.Type` in our database, although I think that should actually be a composite key instead. The `Id` does seem redundant.

This took me down a bit of a rabbit hole to refactor the interface: see https://github.com/bitwarden/server/pull/4877/commits/1e233879bdf6db88310aa5b9e5b485e607f9b44c. It bothered me that the external caller had to retrieve the current `Policy` object, handle a null reference, update it, and _then_ pass it to the command [example](https://github.com/bitwarden/server/blob/6f840758e57e1d24d3a6e81f13a4e6b8df9a0ad5/src/Core/Auth/Services/Implementations/SsoConfigService.cs#L69-L75). This interface also allowed the caller to mess with internal properties like the `Id` or the `CreationDate`, which they shouldn't be able to. I've now refactored it to use an update DTO model so that the caller has both less responsibility and less power.

This also addresses your feedback here because we now match on the organization and type. The update DTO has no concept of `id`.